plugins/USBPrinting/USBPrinterOutputDevice.py

```diff
@@ -127,8 +127,11 @@ def showFirmwareInterface(self):
 
     @pyqtSlot(str)
     def updateFirmware(self, file):
-        # the file path is qurl encoded.
-        self._firmware_location = file.replace("file://", "")
+        # the file path could be qurl encoded.
+        if file[:7] == "file://":
```

Comment:
Minor: I think `file.startswith("file://")` is better.

Reply:
Fixed: 846c55a99d0486bea3623e47412c36d0ab71f0fe